### PR TITLE
Blacklist input, select and textarea from being AMP placeholders

### LINKED ISSUE #3347: Exclude <input> and other input tags from being placeholders
Now that we plan to support forms #3343 we need to make sure AMP differentiate between the AMP `placeholder` attribute to indicate an element is a placeholder for its parent and the `placeholder="some text"` on input-types to show a placeholder text inside the input field.

One way to do this is by just excluding inputs from being AMP placeholders. It doesn't actually make sense since AMP placeholders are mostly used as a waiting to be loaded so I am here until then.

### PR DESCRIPTION
**Pull request by mkhatib**
Closes #3347

ITI: #3343

### CODE REVIEW

File: src/custom-element.js [hunk 1]
```diff
@@ -53,6 +53,13 @@ const PREPARE_LOADING_THRESHOLD_ = 1000;
 
 
 /**
+ * List of tag names of elements that are not allowed to be AMP placeholders.
+ * @private @const {!RegExp}
+ */
+const BLACKLISTED_PLACEHOLDER_TAGS_REGEX_ = /^(?:input|select|textarea)$/i;
```
Comment (aghassemi) [inline src/custom-element.js]: just use a `set`, will be faster than regex
```
const BLACKLISTED_PLACEHOLDER_TAGS = {
 'INPUT': true,
 etc...
}
```

File: src/custom-element.js [hunk 2]
```diff
@@ -53,6 +53,13 @@ const PREPARE_LOADING_THRESHOLD_ = 1000;
...
+
+
```
Comment (mkhatib) [inline src/custom-element.js]: That sounds like a good idea. Went with this approach.

File: src/custom-element.js (hunk 1 above)
Comment (mkhatib) [inline src/custom-element.js]: Went with @sriramkrish85 suggestion of just checking if the element natively have `placeholder` property. So dropped the blacklist.

File: src/custom-element.js [hunk 3]
```diff
@@ -1131,7 +1131,9 @@ function createBaseAmpElementProto(win) {
    * @package @final @this {!Element}
    */
   ElementProto.getPlaceholder = function() {
-    return dom.lastChildElementByAttr(this, 'placeholder');
+    return dom.lastChildElement(this, el => {
+      return el.hasAttribute('placeholder') && !('placeholder' in el);
```
Comment (aghassemi) [inline src/custom-element.js]: nit: Comment what !('placeholder' in el) is doing and why we are doing it

**Comment by aghassemi** [comment]
one nit comment
LGTM

File: src/custom-element.js (hunk 3 above)
Comment (sriramkrish85) [inline src/custom-element.js]: checking if placeholder is a supported attribute by default on that element.

Comment (sriramkrish85) [inline src/custom-element.js]: Ali, i misread your comment. +1 for adding comments

Comment (mkhatib) [inline src/custom-element.js]: Done.

**Pull request merged**